File: Codigo/offline/api/models/Place.py
from db.db import db

# Model para la tabla 'place'
class Place(db.Model):
    id = db.Column(db.Integer, primary_key=True)
    name = db.Column(db.Text)
    abbreviation = db.Column(db.Text)
    description = db.Column(db.Text)
    openTime = db.Column(db.Text)
    closeTime = db.Column(db.Text)
    isActive = db.Column(db.Integer)
    createDate = db.Column(db.Text)

File: Codigo/frontend/api/services/place.ts
import axios from 'axios';
import { Lugar } from '@/api/model/interfaces';
import { URL } from '@/api/constantes'
const BASE_URL = `${URL}/place`;
import { getAdmDni } from './storage';

export async function createLugar(name: string, abbreviation: string, description: string, openTime: string, closeTime: string): Promise<number> {
    try {
        const admDni = await getAdmDni();
        const response = await axios.post(BASE_URL,{
            "name": name,
            "abbreviation": abbreviation,
            "description": description,
            "openTime": openTime,
            "closeTime": closeTime,
            "adm_dni":admDni
        },{
            headers: {
                'Content-Type': 'application/json'
            }
        });
        return response.status;
    } catch (error) {
        console.error('Error al crear lugar: ', error);
        return 400
    }
}